### PR TITLE
#164 implement custom object syncer

### PR DESCRIPTION
**Pull request by lojzatran**
todo: exclude custom objects created by the project-sync from syncing - this will be included in the next PR.

As to the lower code coverage in this PR: the problem is the error and warning callbacks. They're quite hard to test and need extra time. Therefore, I will not address those issues in this PR, but in a different one.

### CODE REVIEW

File: src/main/java/com/commercetools/project/sync/customobject/CustomObjectSyncer.java
Comment (leungkinghin) [inline src/main/java/com/commercetools/project/sync/customobject/CustomObjectSyncer.java:86]: Is here implemented in later?

Comment (lojzatran) [reply src/main/java/com/commercetools/project/sync/customobject/CustomObjectSyncer.java:86]: Check the task description :)

File: src/main/java/com/commercetools/project/sync/customobject/CustomObjectSyncer.java
Comment (leungkinghin) [inline src/main/java/com/commercetools/project/sync/customobject/CustomObjectSyncer.java:105]: We know there should have no update actions in CO. Should we explicitly pass Collections.emptyList() here to override what caller given?

File: src/test/java/com/commercetools/project/sync/customobject/CustomObjectSyncerTest.java
Comment (leungkinghin) [inline src/test/java/com/commercetools/project/sync/customobject/CustomObjectSyncerTest.java:50]: Can it be changed to another name? Because they are not `CustomObjectDraft`

File: src/test/java/com/commercetools/project/sync/customobject/CustomObjectSyncerTest.java
Comment (leungkinghin) [inline src/test/java/com/commercetools/project/sync/customobject/CustomObjectSyncerTest.java:60]: Here it looks confusing. It is better to change another variable name.

File: src/test/java/com/commercetools/project/sync/customobject/CustomObjectSyncerTest.java
Comment (leungkinghin) [inline src/test/java/com/commercetools/project/sync/customobject/CustomObjectSyncerTest.java:59]: Here it looks confusing. It is better to change another variable name.

File: src/test/java/com/commercetools/project/sync/customobject/CustomObjectSyncerTest.java
Comment (lojzatran) [reply src/test/java/com/commercetools/project/sync/customobject/CustomObjectSyncerTest.java]: https://github.com/commercetools/commercetools-project-sync/pull/167/commits/d015858e9467cf0cc68e6dd1387a200b85ba1cdc

File: src/test/java/com/commercetools/project/sync/customobject/CustomObjectSyncerTest.java
Comment (lojzatran) [reply src/test/java/com/commercetools/project/sync/customobject/CustomObjectSyncerTest.java]: https://github.com/commercetools/commercetools-project-sync/pull/167/commits/d015858e9467cf0cc68e6dd1387a200b85ba1cdc

**Review by leungkinghin — APPROVED** [review]

**Comment by lojzatran** [comment]
I will not merge this PR to master for now. This PR will be used instead as a mother PR for all other children PRs related to the custom object sync. Once all other PRs are merged here, it will then be merged to master.

**Pull request merged**
